realtek: add support for D-Link DGS-1210-20

plappermaul opened the pull request:
Hardware specification
 ----------------------

 * RTL8382M SoC, 1 MIPS 4KEc core @ 500MHz
 * 128MB DRAM
 * 32MB NOR Flash
 * 16 x 10/100/1000BASE-T ports
    - Internal PHY with 8 ports (RTL8218B)
    - External PHY with 8 ports (RTL8218B)
 * 4 x Gigabit RJ45/SFP Combo ports
    - External PHY with 4 SFP ports (RTL8214FC)
 * Power LED
 * Reset button on front panel
 * UART (115200 8N1) via unpopulated standard 0.1" pin header marked J6

 UART pinout
 -----------

```
  [o]ooo|J6
   | ||`------ GND
   | |`------- RX
   | `-------- TX
   `---------- Vcc (3V3)
```

 Boot initramfs image from U-Boot
 --------------------------------

  1. Press Escape key during `Hit Esc key to stop autoboot` prompt
  2. Press CTRL+C keys to get into real U-Boot prompt
  3. Init network with `rtk network on` command
  4. Load image with `tftpboot 0x8f000000 openwrt-rtl838x-generic-d-link_dgs-1210-20-initramfs-kernel.bin` command
  5. Boot the image with `bootm` command

To install, upload the sysupgrade image to the OEM webpage or sysupgrade
from the system running from initramfs image.

It has been developed and tested on device with F1 revision.

Signed-off-by: Markus Stockhausen <markus.stockhausen@gmx.de>

Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

musashino205 commented:
Please don't close the PR and open a new one when you update the commit(s)
https://openwrt.org/docs/guide-developer/frequent-pr-mistakes#do_not_multi-post_or_re-post_prs

And there are some issues:

- the prefix ```rtl838x: ``` is wrong, please use the actual target name, ```realtek: ``` or ```realtek: rtl838x: ```
- author name and e-mail address in the [```From: ``` filed of the commit](https://patch-diff.githubusercontent.com/raw/openwrt/openwrt/pull/10067.patch) don't match the name and address in the Signed-off-by line in the commit message

plappermaul commented:
> Please don't close the PR and open a new one when you update the commit(s) https://openwrt.org/docs/guide-developer/frequent-pr-mistakes#do_not_multi-post_or_re-post_prs
> 
> And there are some issues:
> 
> * the prefix `rtl838x: ` is wrong, please use the actual target name, `realtek: ` or `realtek: rtl838x: `
> * author name and e-mail address in the [`From: ` filed of the commit](https://patch-diff.githubusercontent.com/raw/openwrt/openwrt/pull/10067.patch) don't match the name and address in the Signed-off-by line in the commit message
>   
>   * update user.name and user.email properties, or use `--author="name <address>"` option for `git commit`  on your Git CLI.

Better now?

plappermaul committed:
realtek: add support for D-Link DGS-1210-20

 Hardware specification
 ----------------------

 * RTL8382M SoC, 1 MIPS 4KEc core @ 500MHz
 * 128MB DRAM
 * 32MB NOR Flash
 * 16 x 10/100/1000BASE-T ports
    - Internal PHY with 8 ports (RTL8218B)
    - External PHY with 8 ports (RTL8218B)
 * 4 x Gigabit RJ45/SFP Combo ports
    - External PHY with 4 SFP ports (RTL8214FC)
 * Power LED
 * Reset button on front panel
 * UART (115200 8N1) via unpopulated standard 0.1" pin header marked J6

 UART pinout
 -----------

  [o]ooo|J6
   | ||`------ GND
   | |`------- RX
   | `-------- TX
   `---------- Vcc (3V3)

 Boot initramfs image from U-Boot
 --------------------------------

  1. Press Escape key during `Hit Esc key to stop autoboot` prompt
  2. Press CTRL+C keys to get into real U-Boot prompt
  3. Init network with `rtk network on` command
  4. Load image with `tftpboot 0x8f000000 openwrt-rtl838x-generic-d-link_dgs-1210-20-initramfs-kernel.bin` command
  5. Boot the image with `bootm` command

To install, upload the sysupgrade image to the OEM webpage or sysupgrade
from the system running from initramfs image.

It has been developed and tested on device with F1 revision.

Signed-off-by: Markus Stockhausen <markus.stockhausen@gmx.de>

svanheule commented:
Thanks! Merged to master after updating the name of the initramfs file. Commit 6153c530cc2ae139c6cdde68a4ba53bd2f82bf68
